src/components/Button/Button.jsx

```diff
@@ -0,0 +1,138 @@
+import { calculateWPM, handleWordReset } from "../WPM/Wpm";
+import PropTypes from 'prop-types';
+import RiseLoader from "react-spinners/RiseLoader";
+import style from "./Button.module.css";
+import KeyboardLayout from '../Keyboard/KeyboardLayout';
+
+const SPECIAL_KEYS = {
+  SPACE: ' ',
+  CONTROL: 'Control',
+  BACKSPACE: 'Backspace'
+};
+
+function Button({ word, wpm, setword, loading }) {
+  let typedText = "";
+  let startTime;
+
+  const handleKeyPress = (event) => {
+    if (!startTime) {
+      startTime = performance.now();
+    }
+
+    const targetValueInput = document.querySelector("#typeWords");
+    if (!targetValueInput) return; // Early return if target input doesn't exist
+
+    const keyIdentify = event.key;
+    const sanitizedKey = getSanitizedKey(keyIdentify);
+    
+    updateKeyboardVisuals(sanitizedKey);
+    handleTyping(keyIdentify, targetValueInput, event);
+  };
+
+  const getSanitizedKey = (key) => {
+    let sanitized = key.replace(/([.*+?^=${}()`|[\]/\\-])/g, '\\$1');
+    if (key === SPECIAL_KEYS.SPACE) return 'Space';
+    if (key === SPECIAL_KEYS.CONTROL) return 'Ctrl';
+    return sanitized;
+  };
+
+  const updateKeyboardVisuals = (key) => {
+    const isSpecialKey = ["Ctrl", "Shift", "Alt"].includes(key);
+    const target = document.querySelector(`#${key.toLowerCase()}`);
+    
+    if (isSpecialKey) {
+      const leftKey = document.querySelector(`#L${key.toLowerCase()}`);
+      applyKeyPressEffect(target, leftKey);
+    } else {
+      applyKeyPressEffect(target);
+    }
+  };
+
+  const applyKeyPressEffect = (target, paired = null) => {
+    if (!target) return;
+    
+    const removeAllEffects = () => {
+      document.querySelectorAll(`.${style.firstbutton}`).forEach(btn => {
+        btn.classList.remove(style.newbutton);
+      });
+    };
+
+    const addEffect = () => {
+      removeAllEffects();
+      target.classList.add(style.newbutton);
+      if (paired) paired.classList.add(style.newbutton);
+    };
+
+    addEffect();
+    setTimeout(() => target.classList.remove(style.newbutton), 150);
```

Comment:
1. The background color of the button when click is not apply
2. while the code is successfully apply button animation I wanted to keep the class `newbutton` so that the background color of the clicked button is different from the one that is not.
```suggestion
    const addEffect = () => {
      //removeAllEffects();
      target.classList.add(style.newbutton);
      if (paired) paired.classList.add(style.newbutton);
    };

    setTimeout(() => addEffect(), 150);
    //setTimeout(() => target.classList.remove(style.newbutton), 150);
    // only remove the class before add the class back to show animation
    if (target.classList.contains(style.newbutton)) {
      target.classList.remove(style.newbutton);
      if (paired) {
        paired.classList.remove(style.newbutton);
      }
    }
```